### PR TITLE
Add ability to remove avatars

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/8456
Fixes https://github.com/vector-im/riot-web/issues/11287

Based on
image

🎈 
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/components/views/elements/ToggleSwitch.js [hunk 1]
```diff
@@ -32,12 +32,12 @@ const ToggleSwitch = ({checked, disabled=false, onChange, ...props}) => {
...
 
     const _onKeyDown = (e) => {
-        e.stopPropagation();
-        e.preventDefault();
         if (disabled) return;
 
         if (e.keyCode === KeyCode.ENTER || e.keyCode === KeyCode.SPACE) {
             onChange(!checked);
+            e.stopPropagation();
+            e.preventDefault();
```
Comment (turt2live) [inline src/components/views/elements/ToggleSwitch.js]: unrelated changes should probably be in their own PR for next time

File: src/components/views/room_settings/RoomProfileSettings.js [hunk 2]
```diff
@@ -62,13 +62,18 @@ export default class RoomProfileSettings extends React.Component {
...
+    _uploadAvatar = () => {
         this._avatarUpload.current.click();
     };
 
+    _removeAvatar = () => {
+        this.setState({
+            avatarUrl: null,
+            avatarFile: null,
+            enableProfileSave: true,
+        });
```
Comment (turt2live) [inline src/components/views/room_settings/RoomProfileSettings.js]: Don't you need a saveProfile call here?

File: src/components/views/settings/AvatarSetting.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import {ContextMenu, ContextMenuButton, toRightOf, useContextMenu} from "../../structures/ContextMenu";
+import sdk from "../../../index";
+
+// TODO: Why is rendering a box with an overlay so complicated? Can the DOM be reduced?
+const AvatarSetting = (props) => {
```
Comment (turt2live) [inline src/components/views/settings/AvatarSetting.js]: ooi why a function with a bunch of bootstrap instead of a tiny PureComponent class?

File: src/components/views/settings/ProfileSettings.js [hunk 4]
```diff
@@ -52,13 +53,18 @@ export default class ProfileSettings extends React.Component {
...
+    _uploadAvatar = () => {
         this._avatarUpload.current.click();
     };
 
+    _removeAvatar = () => {
+        this.setState({
+            avatarUrl: null,
+            avatarFile: null,
+            enableProfileSave: true,
+        });
```
Comment (turt2live) [inline src/components/views/settings/ProfileSettings.js]: needs a profileSave() call?

File: src/components/views/settings/AvatarSetting.js (hunk 3 above)
Comment (t3chguy) [reply src/components/views/settings/AvatarSetting.js]: Ability to use `useContextMenu` hook simplifies things

File: src/components/views/settings/ProfileSettings.js (hunk 4 above)
Comment (t3chguy) [reply src/components/views/settings/ProfileSettings.js]: Nope, profile is saved when you click the Save button, thus `enableProfileSave`

File: src/components/views/room_settings/RoomProfileSettings.js (hunk 2 above)
Comment (t3chguy) [reply src/components/views/room_settings/RoomProfileSettings.js]: Nope, profile is saved when you click the Save button, thus `enableProfileSave`

File: src/components/views/elements/ToggleSwitch.js (hunk 1 above)
Comment (t3chguy) [reply src/components/views/elements/ToggleSwitch.js]: Yup

File: src/components/views/settings/ProfileSettings.js (hunk 4 above)
Comment (turt2live) [reply src/components/views/settings/ProfileSettings.js]: ah right, this is one of the areas where we don't live-apply changes for some reason.

**Comment by t3chguy** [comment]
Superseded by https://github.com/matrix-org/matrix-react-sdk/pull/3733
